File: CRUDAPI/Models/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace CRUDAPI.Models
{
    public class Task
    {
        [Key]
        public int TaskId { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public bool IsComplete { get; set; }

        // ForeignKey for User
        [ForeignKey("UserId")]
        public int UserId { get; set; }
        
    }
}

File: Lista-De-Tarefas/src/app/Task.ts
export class Task {
    taskId: number;
    title: string;
    description: string;
    isComplete: boolean;
    userId: number;
    // Esse construtor é necessário para criar uma task vazia para usar como base para algumas operações
    constructor(task: any) {
        this.taskId = task.taskId;
        this.title = task.title;
        this.description = task.description;
        this.isComplete = task.isComplete;
        this.userId = task.userId;
      }
}

